File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = dict()

        freq_list = [[] for _ in range(len(nums) + 1)]

        for num in nums:
            count[num] = count.get(num, 0) + 1
        
        for num, val in count.items():
            freq_list[val].append(num)

        res = []
        
        for i in range(len(freq_list) - 1, 0, -1):
            for c in freq_list[i]:
                res.append(c)
                k -= 1
            if k == 0:
                break
        return res
                

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    vector<int> SortVec(vector<int>& v){
        sort(v.begin(), v.end());
        auto it = unique(v.begin(), v.end());
        v.erase(it, v.end());
        return v;
    }

    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()){
            return 0;
        }
        else if (nums.size() == 1){
            return 1;
        }
        else {
            int count = 1;
        vector<int> v;
        nums = SortVec(nums);
        for (int i = 0; i < nums.size() - 1; i++) {
        if (nums[i] + 1 == nums[i + 1] ) {
            count++;
        }
        else {
            v.push_back(count);
            count = 1;
            }
        }
        v.push_back(count);
        int max_elem = 0;
        for (const auto& num : v) {
            if (max_elem < num) {
                max_elem = num;
            }
        }
            return max_elem;
        }
    }
};
